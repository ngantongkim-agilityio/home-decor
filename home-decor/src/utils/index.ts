export * from './products';
export * from './notifications';
