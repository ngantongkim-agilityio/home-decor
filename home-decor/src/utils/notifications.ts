import * as Notifications from 'expo-notifications';

export const requestPermissionsNotifications = async () => {
  await Notifications.requestPermissionsAsync();
  Notifications.setNotificationHandler({
    handleNotification: async () => ({
      shouldShowAlert: true,
      shouldPlaySound: true,
      shouldSetBadge: false,
    }),
  });
};

export const scheduleNotification = async ({
  id = '',
  title = `You've got a new notification! ðŸ“¬`,
  body = 'Here is the notification message',
  seconds = 5,
}) =>
  await Notifications.scheduleNotificationAsync({
    content: { title, body, data: { url: `/details/${id}` } },
    trigger: {
      seconds,
      repeats: false,
    } as Notifications.TimeIntervalTriggerInput,
  });
