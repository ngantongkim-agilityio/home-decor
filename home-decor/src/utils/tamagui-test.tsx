import { PropsWithChildren, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { TamaguiProvider } from 'tamagui';
import { tamaguiConfig } from '@/themes/tamagui.config';

const Providers = ({ children }: PropsWithChildren) => {
  return <TamaguiProvider config={tamaguiConfig}>{children}</TamaguiProvider>;
};

const baseRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react-native';
export { baseRender as render };
