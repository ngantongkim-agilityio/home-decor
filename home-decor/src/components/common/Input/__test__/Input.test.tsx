import { render, fireEvent } from '@/utils/tamagui-test';
import { Input } from '..';

describe('Input Component', () => {
  const props = {
    id: 'test-input',
  };

  it('Should render to match snapshot', () => {
    const component = render(<Input {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('Should render error message', () => {
    const { getByText } = render(
      <Input errorMessage="Invalid input" {...props} />,
    );

    expect(getByText('Invalid input')).toBeTruthy();
  });

  it('Should call onChangeText when input changes', () => {
    const mockOnChangeText = jest.fn();

    const { getByPlaceholderText } = render(
      <Input
        value=""
        placeholder="Enter text"
        onChangeText={mockOnChangeText}
        {...props}
      />,
    );

    const input = getByPlaceholderText('Enter text');
    fireEvent.changeText(input, 'Changed text');

    expect(mockOnChangeText).toHaveBeenCalledWith('Changed text');
  });
});
