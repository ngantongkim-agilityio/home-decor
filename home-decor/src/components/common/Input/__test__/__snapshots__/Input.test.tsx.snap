// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component Should render to match snapshot 1`] = `
<View
  style={[]}
>
  <TextInput
    aria-label=""
    focusVisibleStyle={{}}
    focusable={true}
    id="test-input"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    placeholder=""
    placeholderTextColor="#c0c0c0"
    readOnly={false}
    style={
      {
        "backgroundColor": "#f0e1d2",
        "borderBottomColor": {
          "dynamic": {
            "dark": "hsla(0, 0%, 16%, 1)",
            "light": "hsla(0, 0%, 88%, 1)",
          },
        },
        "borderBottomLeftRadius": 50,
        "borderBottomRightRadius": 50,
        "borderBottomWidth": 0,
        "borderLeftColor": {
          "dynamic": {
            "dark": "hsla(0, 0%, 16%, 1)",
            "light": "hsla(0, 0%, 88%, 1)",
          },
        },
        "borderLeftWidth": 0,
        "borderRightColor": {
          "dynamic": {
            "dark": "hsla(0, 0%, 16%, 1)",
            "light": "hsla(0, 0%, 88%, 1)",
          },
        },
        "borderRightWidth": 0,
        "borderStyle": "solid",
        "borderTopColor": {
          "dynamic": {
            "dark": "hsla(0, 0%, 16%, 1)",
            "light": "hsla(0, 0%, 88%, 1)",
          },
        },
        "borderTopLeftRadius": 50,
        "borderTopRightRadius": 50,
        "borderTopWidth": 0,
        "color": {
          "dynamic": {
            "dark": "hsla(0, 0%, 100%, 1)",
            "light": "hsla(0, 0%, 9%, 1)",
          },
        },
        "fontFamily": "$body",
        "fontSize": 14,
        "height": 41,
        "minWidth": 0,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
  />
</View>
`;
